src/Microsoft.Extensions.DependencyInjection/ServiceLookup/InstanceService.cs

```diff
@@ -27,6 +27,8 @@ public ServiceLifetime Lifetime
 
         public Type ServiceType => Descriptor.ServiceType;
 
+        public Type ImplementationType => Descriptor.ImplementationType;
```

Comment:
Same confusing property since ImplementationType is always null. This should probably be Descriptor.ImplementationInstance.GetType().